pkg/cmd/get.go

```diff
@@ -0,0 +1,162 @@
+	o := NewGetOptions()
+
+	cmd := &cobra.Command{
+		Use:   "get [RESOURCES]",
+		Short: "Command to get resources from KubeArchive",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			err = o.Complete(args)
+			if err != nil {
+				return fmt.Errorf("error completing the args: %w", err)
+			}
+
+			err = o.Run()
+			if err != nil {
+				return fmt.Errorf("error running the command: %w", err)
+			}
+
+			return nil
+		},
+	}
+
+	o.kubeFlags.AddFlags(cmd.Flags())
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().StringVar(&o.KubeArchiveHost, "kubearchive-host", o.KubeArchiveHost, fmt.Sprintf("Host where the KubeArchive API Server is listening. Defaults to '%s'", o.KubeArchiveHost))
+
+	return cmd
+}
+
+func (o *GetOptions) Complete(args []string) error {
+	o.Resource = args[0]
+	if strings.HasPrefix(o.Resource, "v1") {
+		o.IsCoreResource = true
+	}
+
+	o.APIPath = fmt.Sprintf("/apis/%s", o.Resource)
+	if o.IsCoreResource {
+		o.APIPath = fmt.Sprintf("/api/%s", o.Resource)
+	}
+
+	config, err := o.kubeFlags.ToRESTConfig()
+	if err != nil {
+		return fmt.Errorf("error creating the REST configuration: %w", err)
+	}
+
+	o.RESTConfig = config
+
+	return nil
+}
+
+// TODO: when our API returns a List resource, we can remove the decode parameter
+func (o *GetOptions) getResources(host string, decode func([]byte) ([]unstructured.Unstructured, error)) ([]unstructured.Unstructured, error) {
+	url := fmt.Sprintf("%s%s", host, o.APIPath)
+
+	client, err := rest.HTTPClientFor(o.RESTConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating the HTTP client from the REST config: %w", err)
+	}
+	response, err := client.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error on GET to '%s': %w", url, err)
+	}
+	defer response.Body.Close()
+
+	bodyBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error deserializing the body: %w", err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("GET to '%s' returned with code '%d' and body: %s", url, response.StatusCode, string(bodyBytes))
+	}
+
+	return decode(bodyBytes)
+}
+
+func (o *GetOptions) getClusterResources() ([]unstructured.Unstructured, error) {
+	// TODO: when our API returns a List resource, we can remove the inlined function below
+	return o.getResources(o.RESTConfig.Host, func(bodyBytes []byte) ([]unstructured.Unstructured, error) {
+		var list unstructured.UnstructuredList
+		err := json.Unmarshal(bodyBytes, &list)
+		if err != nil {
+			return nil, fmt.Errorf("error deserializing the body into unstructured.UnstructuredList: %w", err)
+		}
+		return list.Items, nil
+	})
+}
+
+func (o *GetOptions) getKubearchiveResources() ([]unstructured.Unstructured, error) {
```

Comment:
shouldn't this be `getKubeArchiveResources` with the `A` capitalized?

Reply:
Fixed!